src/stylesheets/components/_navigation.scss

```diff
@@ -1,59 +1,35 @@
 /*--- Navigation Main, Letter & Number ---*/
-.navigation--main,
-.navigation--letter,
-.navigation--number {
-  ol, ul {
-    @include list--inline;
-  }
-  li {
-    margin-right: 0;
-  }
-  a, span {
+.navigation--main ul,
+.navigation--letter ol,
+.navigation--number ol {
+  @include list--inline;
+
+  span {
     display: block;
-    position: relative;
-    padding: ($base-spacing-unit-tiny / 2) $base-spacing-unit-tiny;
   }
 }
 
 /*--- Navigation Main ---*/
 
-.navigation--main {
-  ol, ul {
-    margin-left: (- $base-spacing-unit);
-  }
-  li {
-    padding: 0 ($base-spacing-unit - 3);
-  }
-}
-
-/*--- Navigation Letter & Number ---*/
+.navigation--main ul {
+  margin: 0 (-$base-spacing-unit-medium);
```

Comment:
By moving the border style from `.navigation--main .container` in here, you can set the margin to zero.